dashboard/app/controllers/plc/user_course_enrollments_controller.rb

```diff
@@ -16,7 +16,12 @@ def create
     if @user_course_enrollment.save
       redirect_to action: :index
     else
-      redirect_to action: :new
+      @user_course_enrollment = Plc::UserCourseEnrollment.find_by(user: @user_course_enrollment.user, plc_course: @user_course_enrollment.plc_course)
```

Comment:
doesn't look like you are using the instance var here, so you could simplify the condition to just check the return value. Or just use `exists?`.